### PR TITLE
add a checkbox for userconsent in discord authorization page

### PR DESCRIPTION
**Pull request by RitikJaiswal75**
This Pr adds a checkbox that takes in user consent to change their nickname in RDS discord server.

Submitting before checking the checkbox
image

button enabled after checking the checkbox
image

### CODE REVIEW

**Review by satyam73 — APPROVED** [review]
LGTM👍

**Review by satyam73 — APPROVED** [review]

File: app/controllers/discord.js [hunk 1]
```diff
@@ -10,24 +10,33 @@ export default class DiscordController extends Controller {
...
-      if (response.status === 204) {
-        this.linkStatus = 'linked';
+        if (response.status === 204) {
+          this.linkStatus = 'linked';
+        } else {
+          this.linkStatus = 'failure';
+        }
       } else {
-        this.linkStatus = 'failure';
+        alert('Please provide your consent by clicking the checkbox');
```
Comment (rohan09-raj) [inline app/controllers/discord.js:39]: Can we please use toast service here ? Check other files for reference where it is being used

**Review by rohan09-raj — CHANGES_REQUESTED** [review]

File: app/controllers/discord.js (hunk 1 above)
Comment (RitikJaiswal75) [reply app/controllers/discord.js]: done

File: app/controllers/discord.js [hunk 2]
```diff
@@ -10,24 +11,37 @@ export default class DiscordController extends Controller {
...
-        this.linkStatus = 'linked';
+        if (response.status === 204) {
+          this.linkStatus = 'linked';
+        } else {
+          this.linkStatus = 'failure';
+        }
       } else {
-        this.linkStatus = 'failure';
+        this.toast.error(
+          'Please provide the consent by clicking the checkbox',
```
[the remark this answers is not in the document]
Comment (RitikJaiswal75) [reply app/controllers/discord.js]: done

**Review by satyam73 — APPROVED** [review]

**Pull request merged**